core/src/main/java/com/datastax/oss/driver/internal/core/metadata/NodeStateEvent.java

```diff
@@ -53,10 +54,10 @@ public static NodeStateEvent removed(DefaultNode node) {
    */
   public final NodeState newState;
 
-  public final DefaultNode node;
+  public final WeakReference<DefaultNode> node;
```

Comment:
Nit: Since we're changing the type here I'm wondering if it might be cleaner to provide a @Nullable getter for DefaultNode, rather than exposing the weak reference directly. Same comment for `DistanceEvent.node`.